### ISSUE TITLE
Old hash no longer accepted

### ISSUE THREAD

**Issue by julien-f** [ranisalt/node-argon2#133]
### Before creating an issue, please be sure to:
- [ ] Checkout to the latest version, **including submodules**
- [ ] Try to find an *isolated* way to reproduce the behavior
- [ ] Fill in all the blanks in the most specific way you can

`argon2` installed via `npm`.

### Steps to reproduce

```js
require('argon2').verify(
  '$argon2i$m=4096,t=3,p=1$tbagT6b1YH33niCo9lVzuA$htv/k+OqWk1V9zD9k5DOBi2kcfcZ6Xu3tWmwEPV3/nc',
  'password'
).then(result => {
  require('assert')(result, 'hash should match password')
})
```

### Expected behaviour

Legacy hashes should still be recognized.

### Actual behaviour

Legacy hashes cannot be verified against their password, the result is `false`.

### Environment
**Operating system**: Fedora 28

**Node version:** 8.11.1

**Compiler version:** gcc 8.1.1

**Comment by ranisalt** [comment]
Version was being parsed as `undefined` and defaulted to the latest. Also check `needsRehash(...)` :)

Published v0.19.2 with the fix.

**Comment by ranisalt** [comment]
Yep, that was requested at #125
